From 6a914ed6d738a9118dccd2f016242446d774378e Mon Sep 17 00:00:00 2001
Message-ID: <6a914ed6d738a9118dccd2f016242446d774378e.1705743628.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1705743627.git.sharlatanus@gmail.com>
References: <cover.1705743627.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 16 Jan 2024 23:42:12 +0000
Subject: [PATCH 19/38] gnu: go-github-com-emersion-go-pgpmail: Move to (gnu
 packages golang-crypto).

* gnu/packages/golang.scm (go-github-com-emersion-go-pgpmail): Move from
here ...
* gnu/packages/golang-crypto.scm: ... to here.

* gnu/packages/mail.scm: Add (gnu packages golang-crypto) to used
modules.

Change-Id: I6e5d284bb045acb30881c79258573c6c79ecc512
---
 gnu/packages/golang-crypto.scm | 31 +++++++++++++++++++++++++++++++
 gnu/packages/golang.scm        | 29 -----------------------------
 gnu/packages/mail.scm          |  1 +
 3 files changed, 32 insertions(+), 29 deletions(-)

diff --git a/gnu/packages/golang-crypto.scm b/gnu/packages/golang-crypto.scm
index 3057e803ba..d1d6122f95 100644
--- a/gnu/packages/golang-crypto.scm
+++ b/gnu/packages/golang-crypto.scm
@@ -2,6 +2,7 @@
 ;;; Copyright © 2018 Pierre Neidhardt <mail@ambrevar.xyz>
 ;;; Copyright © 2019, 2020 Leo Famulari <leo@famulari.name>
 ;;; Copyright © 2021 Raghav Gururajan <rg@raghavgururajan.name>
+;;; Copyright © 2022 (unmatched-parenthesis <paren@disroot.org>
 ;;; Copyright © 2022, 2023 Nicolas Graves <ngraves@ngraves.fr>
 ;;; Copyright © 2023 Artyom V. Poptsov <poptsov.artyom@gmail.com>
 ;;; Copyright © 2023 Clément Lassieur <clement@lassieur.org>
@@ -123,6 +124,36 @@ (define-public go-github-com-aperturerobotics-jacobsa-crypto
 cryptographic standards that are not included in the Go standard library.")
       (license license:asl2.0))))
 
+(define-public go-github-com-emersion-go-pgpmail
+  (package
+    (name "go-github-com-emersion-go-pgpmail")
+    (version "0.2.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/emersion/go-pgpmail")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0ar26b0apw5bxn58qfn1a79cxigbmrqm1irh1rb7x57fydihc7wm"))))
+    (build-system go-build-system)
+    (arguments
+     (list ;; tests don't support our version of protonmail/go-crypto; see
+      ;; <https://github.com/emersion/go-pgpmail/issues/12>
+      #:tests? #f
+      #:import-path "github.com/emersion/go-pgpmail"))
+    (propagated-inputs
+     (list go-github-com-emersion-go-message
+           go-github-com-protonmail-go-crypto
+           go-golang-org-x-crypto
+           go-golang-org-x-text))
+    (home-page "https://github.com/emersion/go-pgpmail")
+    (synopsis "PGP mail encryption for Go")
+    (description "The pgpmail package implements PGP encryption for e-mail
+messages.")
+    (license license:expat)))
+
 (define-public go-github-com-gaukas-godicttls
   (package
     (name "go-github-com-gaukas-godicttls")
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index c8b211441c..b8837b4fcd 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -11748,35 +11748,6 @@ (define-public go-github-com-jhillyerd-enmime
 library geared towards parsing MIME encoded emails.")
     (license license:expat)))
 
-(define-public go-github-com-emersion-go-pgpmail
-  (package
-    (name "go-github-com-emersion-go-pgpmail")
-    (version "0.2.0")
-    (source (origin
-              (method git-fetch)
-              (uri (git-reference
-                    (url "https://github.com/emersion/go-pgpmail")
-                    (commit (string-append "v" version))))
-              (file-name (git-file-name name version))
-              (sha256
-               (base32
-                "0ar26b0apw5bxn58qfn1a79cxigbmrqm1irh1rb7x57fydihc7wm"))))
-    (build-system go-build-system)
-    (arguments
-     (list ;; tests don't support our version of protonmail/go-crypto; see
-           ;; <https://github.com/emersion/go-pgpmail/issues/12>
-           #:tests? #f
-           #:import-path "github.com/emersion/go-pgpmail"))
-    (propagated-inputs (list go-golang-org-x-text
-                             go-golang-org-x-crypto
-                             go-github-com-emersion-go-message
-                             go-github-com-protonmail-go-crypto))
-    (home-page "https://github.com/emersion/go-pgpmail")
-    (synopsis "PGP mail encryption for Go")
-    (description
-     "The pgpmail package implements PGP encryption for e-mail messages.")
-    (license license:expat)))
-
 (define-public go-github-com-gatherstars-com-jwz
   (package
     (name "go-github-com-gatherstars-com-jwz")
diff --git a/gnu/packages/mail.scm b/gnu/packages/mail.scm
index 4466e5e6d9..30ab3b4b76 100644
--- a/gnu/packages/mail.scm
+++ b/gnu/packages/mail.scm
@@ -109,6 +109,7 @@ (define-module (gnu packages mail)
   #:use-module (gnu packages gnupg)
   #:use-module (gnu packages golang)
   #:use-module (gnu packages golang-check)
+  #:use-module (gnu packages golang-crypto)
   #:use-module (gnu packages golang-web)
   #:use-module (gnu packages groff)
   #:use-module (gnu packages gsasl)
-- 
2.41.0

